runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java

```diff
@@ -375,6 +386,60 @@ public void onGameTick(GameTick event)
 		}
 	}
 
+	@Subscribe
+	public void onMenuEntryAdded(final MenuEntryAdded event)
+	{
+		if (!xpTrackerConfig.skillTabOverlayMenuOptions())
+		{
+			return;
+		}
+
+		int widgetIndex = event.getActionParam0();
+		int widgetID = event.getActionParam1();
+		MenuEntry[] menuEntries = client.getMenuEntries();
+
+		if (WidgetInfo.TO_GROUP(widgetID) != WidgetID.SKILLS_GROUP_ID || !event.getOption().startsWith("View"))
```

Comment:
```suggestion
		if (WidgetInfo.TO_GROUP(widgetID) != WidgetID.SKILLS_GROUP_ID || !event.getOption().startsWith("View") || !xpTrackerConfig.skillTabOverlayMenuOptions())
```

Reply:
This is already guarded at the top of the function. These guard checks happen after we get the widget ID from the menu entry. I'll shuffle the variable declarations around to make more sense here.